nad_ch/application/use_cases.py

```diff
@@ -19,19 +26,21 @@ def add_data_provider(ctx: ApplicationContext, provider_name: str) -> None:
     ctx.providers.add(provider)
     ctx.logger.info("Provider added")
 
+    return get_view_model(provider)
```

Comment:
I like this a lot, it seems pretty tidy. I'm wondering, though, if the UI should be the one to call `get_view_model`, rather than the use case. Either way, they should be easily JSON-serializable, right? I like that doing it in the use case keeps the UI super simple, though.

Reply:
Actually, thinking about it more, I feel confident it does belong here, on the more abstract side. 👍